### PR TITLE
Closes #223 Test code clean up

### LINKED ISSUE #223: Cleanup of test code
We have really strong test coverage, but the tests have gotten a little messy in places. I think there's some cleanup we can do.

https://github.com/atorus-research/xportr/pull/200#discussion_r1461830031

https://github.com/atorus-research/xportr/pull/200#discussion_r1461816354

Please add links to other comments as needed for specific issues noticed!

### PR DESCRIPTION
**Pull request by sadchla-codes**
### Thank you for your Pull Request!

We have developed a Pull Request template to aid you and our reviewers. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs. We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

### Changes Description

_(descriptions of changes)_

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description. Can be removed or left blank if changes are minor/self-explanatory.
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Use `styler` package and functions to style files accordingly.
- [x] New functions or arguments follow established convention found in the [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers).
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] The `NEWS.md` entry should go under the `# xportr development version` section. Don't worry about updating the version because it will be auto-updated using the `vbump.yaml` CI.
- [x] Address any updates needed for vignettes and/or templates.
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] The developer is responsible for fixing merge conflicts not the Reviewer.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-metadata.R [hunk 1]
```diff
@@ -759,10 +759,12 @@ test_that("xportr_*: Domain is kept in between calls", {
 # end
 
 test_that("`xportr_metadata()` results match traditional results", {
+
+  if (requireNamespace("magrittr", quietly = TRUE)) {
```
Comment (averissimo) [inline tests/testthat/test-metadata.R:763]: `{magrittr}` is on `Imports` in the `DESCRIPTION` file.

We can safely assume it is installed and remove all of these `requreNamespace("magrittr", ...)`

**Review by bms63 — COMMENTED** [review]
Did you use the admiraldev add-in for this?  If not, maybe we should look into making that a standalone tool

File: tests/testthat/test-metadata.R
Comment (bms63) [inline tests/testthat/test-metadata.R:1]: Do we still need the haven and readxl packages in these tests?

Comment (bms63) [reply tests/testthat/test-metadata.R:1]: Can they be removed so the lintr becomes happy again?

**Review by bms63 — COMMENTED** [review]
Can we just suspend the lintr check for indentation and just have the styler one?

Also can these lines be returned onto the next line where appropriate?

image

**Comment by sadchla-codes** [comment]
> Did you use the admiraldev add-in for this? If not, maybe we should look into making that a standalone tool

I did use admiraldev add-in

File: DESCRIPTION [hunk 2]
```diff
@@ -36,13 +36,13 @@ Imports:
     readr,
     rlang (>= 0.4.10),
     stringr (>= 1.4.0),
-    tidyselect
+    tidyselect,
+    readxl
```
Comment (bms63) [inline DESCRIPTION:40]: where are we using this?  how come it got moved from Suggest to Imports?

**Review by bms63 — CHANGES_REQUESTED** [review]
im weary to move readxl to Imports? why are we doing this?

**Comment by sadchla-codes** [comment]
> im weary to move readxl to Imports? why are we doing this?

Hi @bms63 it was given me a dependency error in the R CMD check, but i made some slight updates. It seems to be working great now and readxl is back to suggest

**Comment by bms63** [comment]
Thanks for organizing and cleaning up our test code.  Much nicer!!!

**Pull request merged**